karma.conf.js

```diff
@@ -1,16 +1,40 @@
 // Karma configuration
 
+const { readFileSync } = require('fs');
 const webpackConfig = require('./webpack.config.test');
 
 // Set BABEL_ENV to use proper preset config
 process.env.NODE_ENV = 'test';
 process.env.BABEL_ENV = 'test';
 
 module.exports = function (config) {
+
+  let specsToRetry;
+  if (config.retryFailed) {
+    // Loads names of failed specs and prepares them for use in a regex.
+    specsToRetry = readFileSync(
+      'test/karma/karma-failed-tests.txt',
+      'utf-8'
+    )
+      .replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
+      .replace(/-/g, '\\x2d')
+      .split('\n')
+      .join('|');
+  }
+
   config.set({
     // base path that will be used to resolve all patterns (eg. files, exclude)
     basePath: '',
 
+    plugins: [
+      'karma-webpack',
+      'karma-jasmine',
+      'karma-sourcemap-loader',
+      'karma-jasmine-html-reporter',
+      'karma-chrome-launcher',
+      require('./test/karma/karma-failed-tests-reporter.js'),
```

Comment:
all these `karma-<name>` plugins were already loaded before automatically, had to add them now that it also loads the `karma-failed-tests` reporter:
https://karma-runner.github.io/6.4/config/plugins.html#loading-plugins